File: tutorials/board-basics/prepare_assets.py
from pathlib import Path
from urllib.request import Request, urlopen

import fitz


root = Path(__file__).parent
user_agent = {"User-Agent": "Papol tutorial asset preparation"}


def download(url: str) -> bytes:
    with urlopen(Request(url, headers=user_agent), timeout=30) as response:
        return response.read()


teapot_png = download("https://commons.wikimedia.org/wiki/Special:Redirect/file/Utah_teapot_2.png?width=1200")
teapot = fitz.open(stream=teapot_png, filetype="png")
teapot[0].get_pixmap(alpha=False).save(root / "utah-teapot.png")

bunny_jpeg = download("https://graphics.stanford.edu/data/3Dscanrep/stanford-bunny-cebal-ssh.jpg")
bunny = fitz.open(stream=bunny_jpeg, filetype="jpeg")
bunny[0].get_pixmap(alpha=False).save(root / "stanford-bunny.png")

File: tutorials/board-basics/prepare-assets.mjs
// The two images the recording drops onto the board, fetched fresh for
// the run and removed by record.mjs afterwards: neither is Papol's to
// keep in the repository.
import { writeFileSync } from 'node:fs';
import path from 'node:path';

const root = path.dirname(new URL(import.meta.url).pathname);
const headers = { 'User-Agent': 'Papol tutorial asset preparation' };

async function download(url, file) {
  const response = await fetch(url, { headers });
  if (!response.ok) throw new Error(`${url}: ${response.status}`);
  writeFileSync(path.join(root, file), Buffer.from(await response.arrayBuffer()));
}

await download('https://commons.wikimedia.org/wiki/Special:Redirect/file/Utah_teapot_2.png?width=1200', 'utah-teapot.png');
await download('https://graphics.stanford.edu/data/3Dscanrep/stanford-bunny-cebal-ssh.jpg', 'stanford-bunny.jpg');
